### PR TITLE
Update media type attributes

### PR DESCRIPTION
**Pull request by JonnyWong16**
## Description

This PR is mostly for bringing forward the attribute changes I made to the media type object from my `3.6.0-tautulli` fork. It also unifies the methods between all the media type objects.

Attribute changes:
* Refactored/added new/updated doc strings for `Audio`, `Artist`, `Album`, and `Track` attributes.
* Refactored/added new/updated doc strings for `Video`, `Movie`, `Show`, `Season`, `Episode`, and `Clip` attributes.
* Added new/updated doc strings for `Photoalbum`, and `Photo` attributes.
* Updated doc strings for `Collection` attributes.
* Updated doc strings for `Playlist` attributes.

New objects:
* Added `media.Style` tag object.

New methods:
* `Track.locations` common property to get the `media.Part` locations of the track.
* `Clip.locations` common property to get the `media.Part` locations of the clip.
* `Photo.locations` common property to get the `media.Part` locations of the photo.
* `Photoalbum.clip(title)` method to return a clip in a photo album matching the `title`.
* `Photoalbum.clips()` method to return all clips in a photo album.
* `Photoalbum.get()` alias to `Photoalbum.photo()`.
* `Collection.item(title)` method to return an item in a collection matching the `title`.
* `Collection.get()` alias to `Collection.item()`.
* `Playlist.item(title)` method to return an item in a playlist matching the `title`.
* `Playlist.get()` alias to `Playlist.item()`.

Method changes:
* Separate `title` and `season` parameter in `Show.season()`.
* New `album` name and `track` number parameters in `Artist.track()`.
* New `track` number parameter in `Album.track()`.

Breaking changes:
* Some `index`, `parentIndex`, `grandparentIndex` and `ratingKey`, `parentRatingKey`, `grandparentRatingKey` attributes were type `str` and have now all been casted to `int` for consistency.
* The `collections` attribute is removed from `Episode` since you cannot add an episode to a collection.
* All `titleSort` attributes now default to `title` instead of returning a blank string.
* ~~The `title` parameter in `Show.season()` used to accept an integer for the season number which is now changed to the `season` parameter.~~ Kept check for `isinstance(title, int)` for backwards compatibility.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [x] I have added tests when applicable

### CODE REVIEW

File: plexapi/library.py [hunk 1]
```diff
@@ -392,6 +387,7 @@ def fetchItems(self, ekey, cls=None, container_start=None, container_size=None,
 
     @property
     def totalSize(self):
+        """ Retruns the total number of item in the library. """
```
Comment (Hellowlol) [inline plexapi/library.py:390]: Typo in Returns and should be items.

File: tests/test_video.py [hunk 2]
```diff
@@ -794,38 +792,35 @@ def test_video_Season_show(show):
...
 
 def test_video_Season_episode(show):
-    episode = show.season(1).get("Winter Is Coming")
+    episode = show.season("Season 1").get("Winter Is Coming")
     assert episode.title == "Winter Is Coming"
 
 
 def test_video_Season_episode_by_index(show):
-    episode = show.season(1).episode(episode=1)
+    episode = show.season("Season 1").episode(episode=1)
```
Comment (Hellowlol) [inline tests/test_video.py:818]: The idea for this was to check that the index works.

**Comment by Hellowlol** [comment]
@JonnyWong16 I’ll merge when I can rebase this. On mobile atm.

**Comment by Hellowlol** [comment]
It was some failing tests. Ill need to manually check these before i merge

**Comment by JonnyWong16** [comment]
Just try re-running it.

File: plexapi/video.py [hunk 3]
```diff
@@ -500,10 +500,11 @@ def season(self, title=None, season=None):
                 :exc:`~plexapi.exceptions.BadRequest`: If title or season parameter is missing.
         """
         key = '/library/metadata/%s/children' % self.ratingKey
-        if title:
+        if title and not isinstance(title, int):
             return self.fetchItem(key, Season, title__iexact=title)
-        elif season:
-            return self.fetchItem(key, Season, index=season)
+        elif season or isinstance(title, int):
```
Comment (JonnyWong16) [inline plexapi/video.py:505]: Just realized that this won't work for `season=0` (specials).

File: plexapi/video.py [hunk 4]
```diff
@@ -613,31 +622,34 @@ def seasonNumber(self):
...
+                :exc:`~plexapi.exceptions.BadRequest`: If title or episode parameter is missing.
         """
-        if not title and not episode:
-            raise BadRequest('Missing argument, you need to use title or episode.')
         key = '/library/metadata/%s/children' % self.ratingKey
         if title:
-            return self.fetchItem(key, title=title)
-        return self.fetchItem(key, parentIndex=self.index, index=episode)
+            return self.fetchItem(key, Episode, title__iexact=title)
+        elif episode:
```
Comment (JonnyWong16) [inline plexapi/video.py:642]: Same here `episode=0` won't work. Probably `elif episode is not None` or `elif isinstance(episode, int)`.

https://i.imgur.com/FrPLgt8.png

**Pull request merged**